src/v/cloud_topics/reconciler/BUILD

```diff
@@ -27,6 +27,9 @@ redpanda_cc_library(
     hdrs = [
         "reconciler.h",
     ],
+    implementation_deps = [
+        "//src/v/kafka/server:partition_proxy",
```

Comment:
Seems a little broken that cloud topics depends on this interface (do they implement it?), but I mean nothing to do here.

Reply:
yeah agreed. i didn't really look into into, but this change should focus any follow-up work on the specific dep.

Reply:
>Seems a little broken that cloud topics depends on this interface (do they implement it?), but I mean nothing to do here.

it's the same dependency that transforms has where we are reading the transaction-normalized data out of a partition. i think bharath improved that some how recently but maybe it isn't its own library yet?

Reply:
Maybe we need a `src/v/kafka/data` or `src/v/kafka/common` that all these places depend on? 🤷

Reply:
Agreed :100: